src/visitors/Tag.js

```diff
@@ -111,6 +131,14 @@ function build(node: Object, context: Context): JSXElement {
 
 const TagVisitor = {
   jsx(node: Object, context: Context) {
+
+    if (getInterpolationRefs(node.name)) {
+      const expr = context.getInterpolationByRef(node.name);
+      if (expr) {
+        return t.jSXExpressionContainer(expr);
+      }
+    }  
```

Comment:
This makes sense. However I'll also have to manage:

```js
const names = pug`
  div
    ${ props.names.map((name) => pug`.name ${name}` }
`
```

The process could be:

1. Check whether the tag with a interpolated ref has attributes or children.
1.1. No attrs or children? Render as jsx expression container (handling interpolated arrays etc..)
2. It does? Check whether the interpolation is a component.
2.1. Not a component? Throw an error.
3. It is? Create jSXElement with associated attrs and children.